### PR TITLE
'caesarsCipher/feature/programAndUI'

### PR DESCRIPTION
**Pull request by IvanParkhomchuk**

### CODE REVIEW

File: caesarsCipher/script.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+const form = document.getElementById('my-form');
+let userInput = document.getElementById('user-input');
+let result = document.getElementById('ciphered-word');
+const copyButton = document.getElementById('copy-btn');
+const copied = document.getElementById('copied');
+
+const TIMEOUT = 3000;
+
+function rot13(str) {
+    let rot13 = {
```
Comment (fantkolja) [inline caesarsCipher/script.js]: Seems like Indian code to me 😂

Please look at Char Codes in javascript String class and String prototype for a more generic implementation.

**Review by fantkolja — CHANGES_REQUESTED** [review]

File: caesarsCipher/script.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  
+    let regex = /\W/;
+    
+    let arr = str.toUpperCase().split('');
+    let newArr = [];
+  
+    for (let i = 0; i < arr.length; i++) {
+        if (regex.test(arr[i])) {
+            newArr.push(arr[i]);
+            continue;
```
Comment (fantkolja) [inline caesarsCipher/script.js]: It's better to use `else` instead of `continue`

File: caesarsCipher/script.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+form.addEventListener('submit', function(e) {
+    e.preventDefault();
+    showResult(rot13(userInput.value));
+});
+
+copyButton.addEventListener('click', function() {
+    navigator.clipboard.writeText(result.textContent);
+    copied.classList.add('move');
+    timer();
+    clearTimeout(timer());
```
Comment (fantkolja) [inline caesarsCipher/script.js]: You call `timer()` two times

**Review by fantkolja — CHANGES_REQUESTED** [review]

**Review by fantkolja — CHANGES_REQUESTED** [review]
If I encode a very long string everything breaks
image

File: caesarsCipher/script.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+const copied = document.getElementById('copied');
+
+const TIMEOUT = 3000;
+
+function rot13(str) {
+    let arr = str.toUpperCase().split('');
+    let newArr = [];
+
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].charCodeAt() < 65 || arr[i].charCodeAt() > 90) {
```
Comment (fantkolja) [inline caesarsCipher/script.js]: Seems like 13, 65, 90 are config variables.
You can move them to constants.

Comment (fantkolja) [reply caesarsCipher/script.js]: To make it even more interesting you can 
- add a new number input on UI, where the user can set the padding for the Cipher. 
- the value should be 13 by default.

File: caesarsCipher/script.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    return newArr.join('');
+}
+
+function timer() {
+    return setTimeout(() => {
+        copied.classList.remove('move');
+    }, TIMEOUT)
+}
+
+function showResult(num) {
```
Comment (fantkolja) [inline caesarsCipher/script.js]: Why is it `num`? Isn't it string?

File: caesarsCipher/script.js (hunk 3 above)
Comment (IvanParkhomchuk) [reply caesarsCipher/script.js]: fixed

File: caesarsCipher/script.js (hunk 2 above)
Comment (IvanParkhomchuk) [reply caesarsCipher/script.js]: fixed

**Comment by IvanParkhomchuk** [comment]
> If I encode a very long string everything breaks image

File: caesarsCipher/script.js (hunk 1 above)
Comment (IvanParkhomchuk) [reply caesarsCipher/script.js]: fixed

File: caesarsCipher/index.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        <link rel="stylesheet" href="../styles/projectsStyle.css">
+    </head>
+    <body>
+        <div class="container">
+            <h1 class="title">CAESARS CIPHER</h1>
+
+            <div class="container-block">
+                <div>
+                    <form class="form" id="my-form">
+                        <p class="text">Enter your word:<input type="text" maxlength="30" class="input" id="user-input"></p>
```
Comment (fantkolja) [inline caesarsCipher/index.html]: What if the requirement is to support the length up to 1000 symbols?
How can we fit large strings on the screen?

**Review by fantkolja — CHANGES_REQUESTED** [review]

File: caesarsCipher/script.js (hunk 4 above)
Comment (IvanParkhomchuk) [reply caesarsCipher/script.js]: fixed

File: caesarsCipher/script.js (hunk 5 above)
Comment (IvanParkhomchuk) [reply caesarsCipher/script.js]: fixed

File: caesarsCipher/script.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    let newArr = [];
+
+    if (padding === 0) padding = 13;
+
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].charCodeAt() < MIN_CHAR || arr[i].charCodeAt() > MAX_CHAR) {
+            newArr.push(arr[i]);
+        }
+        else if (arr[i].charCodeAt() + padding > MAX_CHAR) {
+            newArr.push(String.fromCharCode(arr[i].charCodeAt() - padding));
```
Comment (fantkolja) [inline caesarsCipher/script.js]: I would expect to get "A" as the result of this encoding (Z with 1 padding)
image

**Review by fantkolja — CHANGES_REQUESTED** [review]

File: caesarsCipher/index.html (hunk 6 above)
Comment (IvanParkhomchuk) [reply caesarsCipher/index.html]: fixed

File: caesarsCipher/script.js (hunk 7 above)
Comment (IvanParkhomchuk) [reply caesarsCipher/script.js]: fixed

**Review by fantkolja — APPROVED** [review]

**Pull request merged**